packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -344,6 +344,7 @@ class _PagePosition extends ScrollPositionWithSingleContext implements PageMetri
   Future<void> ensureVisible(
     RenderObject object, {
     double alignment = 0.0,
+    EdgeInsets padding = EdgeInsets.zero,
     Duration duration = Duration.zero,
     Curve curve = Curves.ease,
     ScrollPositionAlignmentPolicy alignmentPolicy = ScrollPositionAlignmentPolicy.explicit,
```

Comment:
I feel like there is opportunity here for this new API to be confused with the `ScrollPositionAlignmentPolicy`. Is there a way that maybe these two could work together?

Reply:
Both `padding` and `alignmentPolicy` work independently of each other. It would be a much larger (and complex) breaking change to alter the semantics of `ScrollPositionAlignmentPolicy` to require individual policies to handle the padding in their own ways.